src/ring_theory/norm.lean

```diff
@@ -0,0 +1,102 @@
+/-
+Copyright (c) 2021 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen
+-/
+
+import linear_algebra.determinant
+import ring_theory.power_basis
+
+/-!
+# Norm for (finite) ring extensions
+
+Suppose we have an `R`-algebra `S` with a finite basis. For each `s : S`,
+the determinant of the linear map given by multiplying by `s` gives information
+about the roots of the minimal polynomial of `s` over `R`.
+
+## Implementation notes
+
+Typically, the norm is defined specifically for finite field extensions.
+The current definition is as general as possible and the assumption that we have
+fields or that the extension is finite is added to the lemmas as needed.
+
+We only define the norm for left multiplication (`algebra.left_mul_matrix`,
+i.e. `algebra.lmul_left`).
+For now, the definitions assume `S` is commutative, so the choice doesn't
+matter anyway.
+
+See also `algebra.trace`, which is defined similarly as the trace of
+`algebra.left_mul_matrix`.
+
+## References
+
+ * https://en.wikipedia.org/wiki/Field_norm
+
+-/
+
+universes u v w
+
+variables {R S T : Type*} [integral_domain R] [integral_domain S] [integral_domain T]
+variables [algebra R S] [algebra R T]
+variables {K L : Type*} [field K] [field L] [algebra K L]
+variables {ι : Type w} [fintype ι]
+
+open finite_dimensional
+open linear_map
+open matrix
+
+open_locale big_operators
+open_locale matrix
+
+namespace algebra
+
+variables (b : basis ι R S)
```

Comment:
Are you actually using this? It seems like all decls are either independent from `b` or explicitly assume a new variable `b` in their statement.